### PR TITLE
Product list scrolling + misc. front-end bug fixes

### LINKED ISSUE #73: Scrolling in the products list is cut off
https://www.loom.com/share/6c9d85096f794ef9bec2add39e5c55e5

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Resolves #73  and various other styling bugs mentioned in #78 comments:
* Fixed product list scrolling cutoff (#73) and converted to FlatList for consistency with everything else changed in #56 
* Changed store names in StoreList to Body (followup from #51 )
* Removed whitespace above the map 

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
See if everything looks right.

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
Smaller store list names
image

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/map/StoreListScreen.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ export default class StoreListScreen extends React.Component {
           keyExtractor={item => item.id}
           // 16px top margin from heading
           ListHeaderComponent={<View style={{ height: 16 }} />}
-          // 400 bottom margin to make sure all search results show with the keyboard activated.
+          // 420 bottom margin to make sure all search results show with the keyboard activated.
```
Comment (kennethlien) [inline screens/map/StoreListScreen.js:110]: blaze it
